dds/DCPS/Serializer.h

```diff
@@ -670,6 +670,7 @@ class OpenDDS_Dcps_Export Serializer {
   bool write_delimiter(size_t size);
 
   enum ConstructionStatus {
+    ConstructionStatusUndefined,
     ConstructionSuccessful,
     ElementConstructionFailure,
     BoundConstructionFailure
```

Comment:
@iguessthislldo @ClaytonCalabrese is there an enumerator from the existing 3 options that should be used as a default value, or should that be a new "Undefined" state?

Reply:
I don't think there should be an "undefined" status unless we change the code that checks the status to error when it encounters that status. Otherwise from my understanding `ConstructionSuccessful` is basically the default state.

Reply:
There are lots of places that seem to set the construction status to ConstructionSuccessful... are those strictly unnecessary? I'm more inclined to put an assert in get_construction_status() to check that the state being returned isn't undefined (since that should never happen, right?).

Reply:
Or are there times where we'd temporarily mark the status as failed but then later may revert to successful?

Reply:
> Or are there times where we'd temporarily mark the status as failed but then later may revert to successful?

Yes. From what I remember it's like an exception. It can be handled in a way that doesn't cause serialization to fail depending on how the user has annotated their IDL.